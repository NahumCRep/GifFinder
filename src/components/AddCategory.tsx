import React, { useState } from "react"
import searchSVG from '../assets/search.svg'

interface Props {
    // setCategoryFn: React.Dispatch<React.SetStateAction<string[]>>
    onAddCategory: (newItem: string) => void
}

const AddCategory = ({ onAddCategory }: Props) => {
    const [searchValue, setSearchValue] = useState<string>('')

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setSearchValue(e.target.value)
    }

    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        if (searchValue.trim().length <= 1) return;
        onAddCategory(searchValue.trim())
        setSearchValue('')
    }

    return (
            <form onSubmit={handleSubmit} aria-label='form'>
                <input
                    role={'searchbox'}
                    aria-label="search-input"
                    type={'text'}
                    value={searchValue}
                    placeholder='Search Gif'
                    onChange={handleChange}
                />
                <button className="submit-btn" type={'submit'}>
                    <img src={searchSVG} alt="search icon" />
                </button>
            </form>
    )
}

export default AddCategory
