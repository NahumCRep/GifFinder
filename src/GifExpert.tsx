import { useState } from 'react'
import GifGrid from './components/GifGrid'
import { AddCategory } from './components'


const GifExpert = () => {
    const [categories, setCategories] = useState<Array<string>>([])

    const handleAdd = (newItem: string) => {
        if (categories.includes(newItem.toLowerCase())) return;
        setCategories(catg => [newItem.toLowerCase(), ...catg])
    }

    return (
        <>
            <header className="header-container">
                <h1 aria-label='title' className='title'>Gif Finder</h1>
                <AddCategory onAddCategory={handleAdd} />
            </header>
            <ol>
                {
                    categories.map(category => (
                        <GifGrid key={category} category={category} />
                    ))
                }
            </ol>
        </>
    )
}

export default GifExpert