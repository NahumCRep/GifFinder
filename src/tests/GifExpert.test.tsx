import { render, screen, fireEvent } from "@testing-library/react";
import GifExpert from "../GifExpert";
import AddCategory from '../components/AddCategory'

describe('Tests for GifExpert Component', () => {
    test('should match snapshot', () => {
        const {container} = render(<GifExpert />)
        expect(container).toMatchSnapshot()
    })
})