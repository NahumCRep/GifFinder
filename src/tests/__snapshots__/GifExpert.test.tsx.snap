// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifExpert Component should match snapshot 1`] = `
<div>
  <header
    class="header-container"
  >
    <h1
      aria-label="title"
      class="title"
    >
      Gif Finder
    </h1>
    <form
      aria-label="form"
    >
      <input
        aria-label="search-input"
        placeholder="Search Gif"
        role="searchbox"
        type="text"
        value=""
      />
      <button
        class="submit-btn"
        type="submit"
      >
        <img
          alt="search icon"
          src="search.svg"
        />
      </button>
    </form>
  </header>
  <main>
    <h2
      class="gifgrid-title"
    />
    <div
      class="card-grid"
    />
    <div
      class="loader-container"
    >
      <div
        class="loader"
        data-testid="loader-spin"
      />
    </div>
  </main>
</div>
`;
